### DISCUSSION TITLE
How to unset a property on an existing database page

### DISCUSSION THREAD

**Discussion by MerlinDE** [takassh/notion-client#8] [Q&A] [answered]
I try to clear/unset properties of a page in a database. 

```rust
let mut props: BTreeMap<String, PageProperty> = BTreeMap::new();
props.insert("meeting date".to_string(), PageProperty::Date { id: None, date: None });
```

Sending this via `UpdatePagePropertiesRequest` basically does nothing:
```rust 
/// p is the page retrieved `QueryDatabaseRequest`

            let mut r = UpdatePagePropertiesRequestBuilder::default();
            r.properties(props);
            match r.build() {
                Ok(req) => {
                    let _ = client
                        .pages
                        .update_page_properties(&*p.id, req)
                        .await;
                }
                Err(e) => {
                    println!("Error: {:#?}", e);
                }
            }
```

I get similar results with other property types, so I guess I'm missing something. :)

**Comment by takassh** [comment, marked as answer]
@MerlinDE 
Hi thank you for opening discussion.
I have investigated this issue and found that we just needed `Option` for the `BTreeMap` value.

I have updated the code. Please update your local version to the latest as well!

**Reply by MerlinDE** [reply]
Awesome! That works. Thank you for the swift reply and fix!
